src/ucp/dt/datatype_iter.c

```diff
@@ -11,6 +11,159 @@
 #include "datatype_iter.inl"
 
 
+#define ucp_datatype_iter_iov_for_each(_index, _length, _dt_iter) \
+    for (_index = 0, _length = 0; _length < (_dt_iter)->length; \
+         _length += (_dt_iter)->type.iov.iov[_index++].length)
```

Comment:
how here prevents situation when _index is out of iov.iov_count?

Reply:
it's because length would reach dt_iter->length before _index raches iov_count. but I will add an assertion